Add plugin: AI Providers

pfrankov opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/pfrankov/obsidian-ai-providers

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

pfrankov commented:
/skip

pfrankov commented:
There is no security risks, since the code is not rely on user's input.

pfrankov commented:
> ### Required
> [[1](https://github.com/pfrankov/obsidian-ai-providers/blob/571c2d1182d9531e89ce35e20577617afb2128eb/src/settings.ts#L279)][[2](https://github.com/pfrankov/obsidian-ai-providers/blob/571c2d1182d9531e89ce35e20577617afb2128eb/src/settings.ts#L310)]:Using `innerHTML`, `outerHTML` or similar API's is a security risk. Instead, use the DOM API or the Obsidian helper functions: https://docs.obsidian.md/Plugins/User+interface/HTML+elements

Fixed both with sanitizeHTMLToDom

joethei commented:
[Copyright (c) 2024 Pavel Frankov](https://github.com/pfrankov/obsidian-ai-providers/blob/dea1fea2ea94a750c0ebda359df2d7daac977d99/LICENSE#L3)
Wrong year.

["main": "dist/main.js"](https://github.com/pfrankov/obsidian-ai-providers/blob/dea1fea2ea94a750c0ebda359df2d7daac977d99/manifest.json#L10)
This is not a thing in the manifest, remove it.

[if (this.isFormOpen && this.editingProvider) {](https://github.com/pfrankov/obsidian-ai-providers/blob/dea1fea2ea94a750c0ebda359df2d7daac977d99/src/settings.ts#L132-L270)
This should be a proper modal.

[iconEl.appendChild(sanitizeHTMLToDom(provider.type === 'openai' ? openAIIcon : ollamaIcon));](https://github.com/pfrankov/obsidian-ai-providers/blob/dea1fea2ea94a750c0ebda359df2d7daac977d99/src/settings.ts#L310)
Use the `setIcon` / `addIcon` API's for this: <https://docs.obsidian.md/Plugins/User+interface/Icons>

pfrankov commented:
@joethei thank you for the review!

> [Copyright (c) 2024 Pavel Frankov](https://github.com/pfrankov/obsidian-ai-providers/blob/dea1fea2ea94a750c0ebda359df2d7daac977d99/LICENSE#L3) Wrong year.
> 
Fixed.

> ["main": "dist/main.js"](https://github.com/pfrankov/obsidian-ai-providers/blob/dea1fea2ea94a750c0ebda359df2d7daac977d99/manifest.json#L10) This is not a thing in the manifest, remove it.
> 
Fixed

> [if (this.isFormOpen && this.editingProvider) {](https://github.com/pfrankov/obsidian-ai-providers/blob/dea1fea2ea94a750c0ebda359df2d7daac977d99/src/settings.ts#L132-L270) This should be a proper modal.
> 
Fixed

> [iconEl.appendChild(sanitizeHTMLToDom(provider.type === 'openai' ? openAIIcon : ollamaIcon));](https://github.com/pfrankov/obsidian-ai-providers/blob/dea1fea2ea94a750c0ebda359df2d7daac977d99/src/settings.ts#L310) Use the `setIcon` / `addIcon` API's for this: https://docs.obsidian.md/Plugins/User+interface/Icons
Removed icons completely.
